suite-common/wallet-core/src/fiat-rates/fiatRatesSelectors.ts

```diff
@@ -0,0 +1,211 @@
+import { A, D, F, pipe } from '@mobily/ts-belt';
+import { memoize, memoizeWithArgs } from 'proxy-memoize';
+
+import { FiatCurrencyCode } from '@suite-common/suite-config';
+import {
+    Account,
+    AccountKey,
+    WalletAccountTransaction,
+    FiatRateKey,
+    FiatRatesStateLegacy,
+    Rate,
+    RateType,
+    TickerId,
+} from '@suite-common/wallet-types';
+import { getFiatRateKeyFromTicker } from '@suite-common/wallet-utils';
+
+import {
+    AccountsRootState,
+    selectAccountByKey,
+    selectDeviceAccounts,
+} from '../accounts/accountsReducer';
+import { TransactionsRootState, selectTransactions } from '../transactions/transactionsReducer';
+import { MAX_AGE, ONE_WEEK_IN_MS } from './fiatRatesConstants';
+import { FiatRatesRootState } from './fiatRatesTypes';
+
+export const selectFiatRatesByFiatRateKey = (
+    state: FiatRatesRootState,
+    fiatRateKey: FiatRateKey,
+    rateType: RateType = 'current',
+): Rate | undefined => state.wallet.fiat?.[rateType]?.[fiatRateKey];
+
+export const selectIsFiatRateLoading = (
+    state: FiatRatesRootState,
+    fiatRateKey: FiatRateKey,
+    rateType: RateType = 'current',
+) => {
+    const currentRate = selectFiatRatesByFiatRateKey(state, fiatRateKey, rateType);
+    return currentRate?.isLoading ?? false;
+};
+
+export const selectIsTickerLoading = (
+    state: FiatRatesRootState,
+    ticker: TickerId,
+    fiatCurrency: FiatCurrencyCode,
+    rateType: RateType = 'current',
+) => {
+    const fiatRateKey = getFiatRateKeyFromTicker(ticker, fiatCurrency);
+    return selectIsFiatRateLoading(state, fiatRateKey, rateType);
+};
+
+export const selectShouldUpdateFiatRate = (
+    state: FiatRatesRootState,
+    fiatRateKey: FiatRateKey,
+    rateType: RateType = 'current',
+) => {
+    const currentRate = selectFiatRatesByFiatRateKey(state, fiatRateKey, rateType);
+
+    if (!currentRate) {
+        return true;
+    }
+
+    const { lastSuccessfulFetchTimestamp } = currentRate;
+    const now = Date.now();
+
+    return now - lastSuccessfulFetchTimestamp > MAX_AGE[rateType];
+};
+
+export const selectTickerFromAccounts = memoize((state: FiatRatesRootState): TickerId[] => {
+    const accounts = selectDeviceAccounts(state as any);
+    return pipe(
+        accounts,
+        A.map(account => [
+            {
+                symbol: account.symbol,
+            } as TickerId,
+            ...(account.tokens || []).map(
+                token =>
+                    ({
+                        symbol: account.symbol,
+                        tokenAddress: token.contract,
+                    }) as TickerId,
```

Comment:
Lets cast only what needs to be casted at the lowest possible level.

```ts
                (token): TickerId => ({
                    symbol: account.symbol,
                    tokenAddress: token.contract as TokenAddress,
                }),
                ```

Reply:
adding `asTokenAddress` utility function would be even better

Reply:
I like to discuss this more, but for now it is not blocking the merge. Feel free to resolve if you really don't want to fix it now.